_posts/2021-04-28-#17331.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: pr
+date: 2021-04-28
+title: "Use effective values throughout coin selection"
+pr: 17331
+authors: [achow101]
+components: ["wallet"]
+host: Xekyo
+status: upcoming
+commit: 4ac1add
+---
+
+## Notes
+
+- Wallets maintain their balance as a pool of Unspent Transaction Outputs
+  (UTXOs).
+
+- _Coin selection_ refers to the process of picking UTXOs from the wallet's UTXO
+  pool to fund a transaction.
+
+- Beyond the primary goal of funding a transaction, the secondary goals of coin
+  selection are:
+
+    - Minimizing short term and long term fees:  
+      We want to both pay the least right now, but we also want to be good
+      stewards of our UTXO pool and not rack up large UTXO counts that will
+      translate to huge future transaction fees.
```

Comment:
Doesn't totally fit the "short term and long term fees" description. I would think that it's more about a user wanting to pay the minimum amount to get their tx confirmed in time. Including long term fees e.g. spending change in the future

Reply:
I've rewritten this section to:

>    - Minimizing short term and long term fees:  
>       We have to balance two needs, we want to pay the smallest fee now to get a
>       timely confirmation, but we have to keep in mind that all of our UTXOs
>       need to be spent in the future. So, we shouldn't over-optimize locally at
>       the detriment of large future costs. E.g. always using largest-first
>       selection minimizes the current cost, but grinds your UTXO pool to dust.

Does that make more sense?

Reply:
yes!!!